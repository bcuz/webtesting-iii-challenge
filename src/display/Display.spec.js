import React from 'react';
import renderer from 'react-test-renderer'; 
import { render, fireEvent } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each'
import 'jest-dom/extend-expect';

import Display from './Display';

describe('<Display />', () => {
  // 2. write this test
  it('matches snapshot default values', () => {
    const tree = renderer.create(<Display />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('matches snapshot with closed gate ', () => {
    const tree = renderer.create(<Display closed={true} />); // generates a DOM tree

    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('matches snapshot with closed and locked gate ', () => {
    const tree = renderer.create(<Display closed={true} locked={true} />); // generates a DOM tree

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
