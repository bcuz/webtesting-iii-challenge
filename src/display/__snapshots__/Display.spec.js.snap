// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> matches snapshot default values 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;

exports[`<Display /> matches snapshot with closed and locked gate  1`] = `
<div
  className="display panel"
>
  <div
    className="led red-led"
  >
    Locked
  </div>
  <div
    className="led red-led"
  >
    Closed
  </div>
</div>
`;

exports[`<Display /> matches snapshot with closed gate  1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led red-led"
  >
    Closed
  </div>
</div>
`;
